File: 2 term/PB/PB Lab2.5/PB Lab2.5 cs/IOManager.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PB_Lab2._5_cs
{
    public class IOManager
    {
        public List<Prism> GetPrisms()
        {
            List<Prism> prisms = new List<Prism>();
            Console.Write("Enter number of prisms you want to enter: ");
            if (int.TryParse(Console.ReadLine(), out int prismCount))
            {
                for (int i = 0; i < prismCount; i++)
                {
                    Console.Write("Enter prism in format: number of sides(3 or 4), side length, heigth: ");
                    string[] input = Console.ReadLine().Split(',');
                    try
                    {
                        if (int.Parse(input[0]) == 3)
                        {
                            prisms.Add(new Prism3(double.Parse(input[1]), double.Parse(input[2])));
                        }
                        else if (int.Parse(input[0]) == 4)
                        {
                            prisms.Add(new Prism4(double.Parse(input[1]), double.Parse(input[2])));
                        }
                        else
                        {
                            Console.WriteLine("Wrong number of sides entered, try inputing this prism again");
                            i--;
                        }
                    }
                    catch (Exception ex)
                    {
                        Console.WriteLine("Something went wrong, try inputing this prism again");
                        i--;
                    }
                }
            }
            return prisms;
        }
        private (double volumeSum, double surfaceSum) ProcessPrisms(List<Prism> prisms)
        {
            double volumeSum = 0;
            double surfaceSum = 0;
            foreach (Prism prism in prisms)
            {
                if (prism is Prism3)
                {
                    volumeSum += prism.CalculateVolume();
                }
                else if (prism is Prism4)
                {
                    surfaceSum += prism.CalculateSurface();
                }
            }
            return (volumeSum, surfaceSum);
        }
        public void GetVolumeSurfaceData(List<Prism> prisms)
        {
            (double volumeSum, double surfaceSum) = ProcessPrisms(prisms);
            Console.WriteLine($"Sum of volumes of triangular prisms: {volumeSum}");
            Console.WriteLine($"Sume of surfaces of square prisms: {surfaceSum}");
        }
    }
}

File: 2 term/PB/PB Lab2.5/PB Lab2.5 py/io_manager.py
from Prism3 import Prism3
from Prism4 import Prism4
from Prism import Prism


def get_prisms():
    prism_count = int(input("Enter number of prisms you want to enter: "))
    i = 0
    prisms = []
    while i < prism_count:
        line = input("Enter a prism in format: side number(3 or 4) side length height: ")
        members = line.split(" ")
        if len(members) == 3:
            if int(members[0]) == 3:
                prisms.append(Prism3(float(members[1]), float(members[2])))
                i += 1
            elif int(members[0]) == 4:
                prisms.append(Prism4(float(members[1]), float(members[2])))
                i += 1
            else:
                print("Wrong number of sides, try again")
        else:
            print("Something went wrong, try again")
    return prisms

def get_volume_surface_data(prisms):
    volume_sum = 0
    surface_sum = 0
    for prism in prisms:
        if type(prism) is Prism3:
            volume_sum += prism.calculate_volume()
        elif type(prism) is Prism4:
            surface_sum += prism.calculate_surface()
    print(f"Sum of volumes of triangular prisms is {volume_sum}")
    print(f"Sum of surfaces of square prisms is {surface_sum}")

